File: app/src/main/java/com/example/strile/service/structures/Subtask.java
package com.example.strile.service.structures;

public class Subtask {
    private final String name;
    private final boolean complete;

    public Subtask(String name, boolean complete) {
        this.name = name;
        this.complete = complete;
    }

    public String getName() {
        return name;
    }

    public boolean isComplete() {
        return complete;
    }

    public Subtask setState(boolean state) {
        return new Subtask(name, state);
    }
}

File: app/src/main/java/com/example/strile/data_firebase/models/Subtask.kt
package com.example.strile.data_firebase.models

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Subtask(var name: String = "", var isComplete: Boolean = false): Parcelable {

    fun setState(state: Boolean): Subtask {
        return Subtask(name, state)
    }
}